packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -290,13 +312,55 @@ function objectToMapTree(data: InsertableContent, schema: TreeNodeSchema): MapTr
 	// Filter keys to only those that are strings - our trees do not support symbol or numeric property keys
 	const keys = Reflect.ownKeys(data).filter((key) => typeof key === "string") as FieldKey[];
 
+	const allKeys = Object.keys((schema as any).info) as FieldKey[];
+	const keysWithoutData = allKeys.filter((value) => !keys.includes(value));
+	for (const key of keysWithoutData) {
+		const fieldSchema = getObjectFieldSchema(schema, key);
+		if (fieldSchema.kind === FieldKind.Identifier) {
+			const fieldValue = nodeKeyManager.stabilizeNodeKey(
+				nodeKeyManager.generateLocalNodeKey(),
+			);
+			const mappedChildTree = nodeDataToMapTree(
+				fieldValue,
+				fieldSchema.allowedTypeSet,
+				nodeKeyManager,
+			);
+			const flexKey: FieldKey = brand(getStoredKey(key, fieldSchema));
+
+			// Note: SchemaFactory validates this at schema creation time, with a user-friendly error.
+			// So we don't expect to hit this, and if we do it is likely an internal bug.
+			assert(!fields.has(flexKey), 0x925 /* Keys must not be duplicated */);
+			fields.set(flexKey, [mappedChildTree]);
+		} else if (fieldSchema.kind === FieldKind.Optional) {
+			const defaultProvider = fieldSchema.props?.defaultProvider;
+			if (defaultProvider !== undefined) {
+				const fieldValue = extractFieldProvider(defaultProvider)(nodeKeyManager);
+				const mappedChildTree = nodeDataToMapTree(
+					fieldValue,
+					fieldSchema.allowedTypeSet,
+					nodeKeyManager,
+				);
+				const flexKey: FieldKey = brand(getStoredKey(key, fieldSchema));
+
+				// Note: SchemaFactory validates this at schema creation time, with a user-friendly error.
+				// So we don't expect to hit this, and if we do it is likely an internal bug.
+				assert(!fields.has(flexKey), 0x925 /* Keys must not be duplicated */);
+				fields.set(flexKey, [mappedChildTree]);
```

Comment:
This logic should be applied regardless of field kind. Delete the special case for identifiers above, and the check for optional fields. Identifiers should always get a default provider, and so should optional fields (you might need to update SchemaFactory.optional to do that)

Reply:
Updated schemaFactory with default default providers, and deleted special cases